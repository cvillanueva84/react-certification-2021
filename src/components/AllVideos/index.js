export { default } from './AllVideos.component';