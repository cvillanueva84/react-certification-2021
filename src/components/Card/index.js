export {default} from './Card.component.jsx'; 