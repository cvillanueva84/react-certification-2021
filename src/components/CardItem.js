import React from 'react';
import { useContext } from 'react';
import { GlobalContext } from './Context';
import {CardItemUi}  from './ui/CardItemUi'

export const CardItem = ({ title, description, url, videoId }) => {

  const { myStateReducer, dispatch } = useContext(GlobalContext);

  const handleClick = () => {
      dispatch({
        type: 'actionAddVideoData',
        payload: {
          ...myStateReducer,
          videoData: {
            title: title,
            description: description,
            url: url,
            videoId: videoId
          }
        }
      });
  }

  return (
    <CardItemUi 
      title={title} 
      description={description} 
      url={url} 
      handleClick={handleClick}
    />
  );

};

