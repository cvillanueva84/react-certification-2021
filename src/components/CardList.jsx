import React from 'react';
import Grid from '@material-ui/core/Grid';
import VideoCard from './VideoCard';
import videos from '../youtube-videos-mock';

export default function CardList() {
  return (
    <Grid
      container
      justifyContent="center"
      alignItems="center"
      style={{ padding: '30px' }}
    >
      {videos.items.map((video) => (
        <VideoCard videoInfo={video.snippet} key={video.etag} />
      ))}
    </Grid>
  );
}
