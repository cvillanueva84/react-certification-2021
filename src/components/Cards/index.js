export { default } from './Cards.component';
