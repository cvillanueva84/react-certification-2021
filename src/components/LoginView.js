import React, { useContext } from 'react';
import { useForm } from '../hooks/useForm';
import { GlobalContext } from './Context';
import { firebase, googleAuthProvider } from '../firebase/firebase-config';
import { useHistory } from 'react-router';

import { LoginViewUi } from './ui/LoginViewUi';


export const LoginView = () => {
  const { myStateReducer, dispatch } = useContext(GlobalContext);
  const history = useHistory();

  const initialForm = {
    textUser: '',
    textPass: '',
  };
  const [formValues, handleInputChange, reset] = useForm(initialForm);
  const { textUser, textPass } = formValues;

  const handlelogin = (e) => {
    e.preventDefault();

    if (textUser === 'Wizeline' && textPass === 'Rocks!') {
      const lastPath = localStorage.getItem('lastPath') || '/';
      dispatch({
        type: 'actionLogin',
        payload: {
          ...myStateReducer,
          user: textUser,
        },
      });
      history.replace(lastPath);
    } else {
      alert('Usuario no valido');
      reset();
    }
  };

  const handleGooglelogin = (e) => {
    e.preventDefault();

    firebase.auth().signInWithPopup(googleAuthProvider)
    .then(({ user }) => {
        dispatch({
            type: 'actionLoginGoogle',
            payload: {
              ...myStateReducer,
              user: user.displayName,
            },
          });
    }).catch((e) => {
        alert(`Algo anda mal: ${e}`);
    });
  }

  return (
    <LoginViewUi 
      handleInputChange={handleInputChange}
      handlelogin={handlelogin}
      handleGooglelogin={handleGooglelogin}
      textUser={textUser}
      textPass={textPass}
    />
  );
};
