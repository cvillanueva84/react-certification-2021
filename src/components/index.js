export { default as Navbar } from './Navbar/Navbar';
export { default as SearchBar } from './SearchBar/SearchBar';
export { default as ThemeSwitch } from './ThemeSwitch/ThemeSwitch';
export { default as Results } from './Results/Results';
