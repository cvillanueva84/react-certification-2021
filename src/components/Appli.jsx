import React from 'react';
import CardList from './CardList';
import NavBar from './NavBar';

export default function Appli() {
  return (
    <div>
      <NavBar />
      <CardList />
    </div>
  );
}
