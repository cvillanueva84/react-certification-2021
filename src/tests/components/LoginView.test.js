import React from 'react';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { GlobalContext } from '../../components/Context';
import { MemoryRouter, Router } from 'react-router';
import { LoginView } from '../../components/LoginView';
import { renderHook } from '@testing-library/react-hooks';
import { useForm } from '../../hooks/useForm';


describe('Test Login View', () => {
    const contextValue = {
        dispatch: jest.fn(),
        myStateReducer: {
            mode: 'light',
            search: 'wizeline',
            videoData: {},
            videoDataFav: {},
            user: '',
            logged: false,
            favorites: [],
        }
    }
    const historyMock = {
        push: jest.fn(),
        replace: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn()
    }

    const wrapper = mount(
        <GlobalContext.Provider value={contextValue}>
            <MemoryRouter>
                <Router history={historyMock}>
                    <LoginView />
                </Router>
            </MemoryRouter>
        </GlobalContext.Provider>
    );

    afterEach(() => {
        jest.clearAllMocks();
    });

    const initialForm = {
        textUser: 'Wizeline',
        textPass: 'Rocks!',
      };

    it('should render success', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.exists());
    });

    it('should Login', () => {
        const e = {preventDefault: jest.fn()};
        window.alert = () => {};
        const { result } =  renderHook(() => useForm(initialForm));
        const [formValues ] = result.current;
        console.log(formValues);
        wrapper.find('button').prop('onClick')(e);
    });

    it('should Login Google', () => {
        window.alert = () => {};
        wrapper.find('img').simulate('click');
    });
    
});