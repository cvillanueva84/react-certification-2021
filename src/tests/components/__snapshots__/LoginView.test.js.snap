// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Login View should render success 1`] = `ReactWrapper {}`;
